<!DOCTYPE html>
<html lang="en"><head>
<script src="HOBI_Bioinfo_Presentation_files/libs/clipboard/clipboard.min.js"></script>
<script src="HOBI_Bioinfo_Presentation_files/libs/quarto-html/tabby.min.js"></script>
<script src="HOBI_Bioinfo_Presentation_files/libs/quarto-html/popper.min.js"></script>
<script src="HOBI_Bioinfo_Presentation_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="HOBI_Bioinfo_Presentation_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="HOBI_Bioinfo_Presentation_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="HOBI_Bioinfo_Presentation_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="HOBI_Bioinfo_Presentation_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.3.353">

  <meta name="author" content="Candidate: Heather Kates">
  <title>Bioinformatics Analyst III</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="HOBI_Bioinfo_Presentation_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="HOBI_Bioinfo_Presentation_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="HOBI_Bioinfo_Presentation_files/libs/revealjs/dist/theme/quarto.css">
  <link href="HOBI_Bioinfo_Presentation_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="HOBI_Bioinfo_Presentation_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="HOBI_Bioinfo_Presentation_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="HOBI_Bioinfo_Presentation_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-titled.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-titled) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-titled.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-titled .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-titled .callout-title  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-titled.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-titled.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-title {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-title {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-title {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-titled .callout-body > .callout-content > :last-child {
    margin-bottom: 0.5rem;
  }

  .callout.callout-titled .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-titled) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-title {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-title {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-title {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-title {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-title {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Bioinformatics Analyst III</h1>
  <p class="subtitle">Biomedical Informatics Program (BMI)</p>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Candidate: Heather Kates 
</div>
</div>
</div>

</section>
<section id="guiding-principles" class="slide level2">
<h2>Guiding principles<br><br></h2>
<p><font size="7"></font></p><font size="7">
</font><ul><font size="7">
<li><p>Reproducibility</p></li>
<li><p>Efficiency</p></li>
</font><li><font size="7"></font><p><font size="7">Creativity </font></p></li>
</ul>
</section>
<section id="highlighted-bioinformatics-projects" class="slide level2">
<h2>Highlighted bioinformatics projects<br><br></h2>
<p><font size="7"></font></p><font size="7">
</font><ol type="1"><font size="7">
<li><p>Population genomics with exome-seq</p></li>
<li><p>Massive phylogenetics with hyb-seq</p></li>
</font><li><font size="7"></font><p><font size="7">End-to-end R pipeline for metabolomics core </font></p></li>
</ol>
</section>
<section id="highlighted-bioinformatics-projects-1" class="slide level2">
<h2>Highlighted bioinformatics projects<br><br></h2>
<p><font size="7"></font></p><font size="7">
<ol type="1">
<li>Population genomics with exome-seq</li>
</ol>
<p><span style="color: gray;">2. Massive phylogenetics with hyb-seq</span></p>
</font><p><font size="7"><span style="color: gray;">3. End-to-end R pipeline for metabolomics core </span></font></p>
</section>
<section id="population-genomics-with-exome-seq" class="slide level2">
<h2>1. <a href="https://www.frontiersin.org/articles/10.3389/fevo.2021.618380/full#h3">Population genomics with exome-seq</a></h2>
<p><a href="https://www.frontiersin.org/files/Articles/618380/fevo-09-618380-HTML/image_m/fevo-09-618380-g005.jpg">Aim</a>: Identify regions of the genome with reduced genetic diversity in crops relative to wild</p>
<p>Step 1: Identify exome genes to target and sequence</p>
<ul>
<li><p><a href="https://www.sciencedirect.com/science/article/pii/S1055790317301811?casa_token=3n1W3eNlvhgAAAAA:l7kW1KUsTjVHogkSwX862SlReLzT8RbHb4KI9SLsB4CW6U7szoyMghGkewRxmRsWixvzKJ7Cmv8#s0105">Bash pipeline to identify genes to sequence</a></p></li>
<li><p><a href="https://github.com/HeatherKates/Locus_selection_for_RAPiD/tree/master">Mining publicly available genomes for SNPs</a></p></li>
</ul>
<p>Step 2: NGS DNA sequence processing</p>
<p>Step 3: Exome-wide genetic diversity analysis</p>
</section>
<section id="ngs-dna-sequence-processing" class="slide level2">
<h2>NGS DNA sequence processing</h2>
<p><font size="5"></font></p><font size="5">
<table>
<colgroup>
<col style="width: 38%">
<col style="width: 33%">
<col style="width: 27%">
</colgroup>
<thead>
<tr class="header">
<th>Process</th>
<th>Software</th>
<th>Samples</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Barcode splitting, filtering, trimming</td>
<td>FASTX toolkit (bash, hipergator)</td>
<td>96</td>
</tr>
<tr class="even">
<td>Align reads to reference genome</td>
<td>MOSAIK 2.3.2</td>
<td>96</td>
</tr>
<tr class="odd">
<td>SNP calling</td>
<td>FREEBAYES 0.9.15</td>
<td>96</td>
</tr>
<tr class="even">
<td>SNP quality-filtering</td>
<td>VCFtools</td>
<td>96</td>
</tr>
</tbody>
</table>
</font><p><font size="5"></font></p>
</section>
<section id="exome-wide-genetic-diversity-analysis" class="slide level2">
<h2>Exome-wide genetic diversity analysis</h2>
<p><font size="5"></font></p><font size="5">
<table>
<colgroup>
<col style="width: 31%">
<col style="width: 31%">
<col style="width: 36%">
</colgroup>
<thead>
<tr class="header">
<th>Process</th>
<th>Software</th>
<th>Independent executions</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Estimation of population structure</td>
<td>STRUCTURE (bash on hipergator)</td>
<td>2 datasets x 10 burn-ins x 10 <em>K</em> values</td>
</tr>
<tr class="even">
<td>Phylogenetic tree building</td>
<td>SNPhylo (bash on hipergator)</td>
<td>2 data sets</td>
</tr>
<tr class="odd">
<td>Genetic diversity calculation and GO enrichment analysis</td>
<td>4P (bash on hipergator), AmiGo (RamiGO in R)</td>
<td>2 data sets</td>
</tr>
<tr class="even">
<td>Map genes to chromosome position</td>
<td>PopGenome (R)</td>
<td>2 data sets</td>
</tr>
</tbody>
</table>
</font><p><font size="5"></font></p>
</section>
<section id="highlighted-bioinformatics-projects-2" class="slide level2">
<h2>Highlighted bioinformatics projects<br><br></h2>
<p><font size="7"></font></p><font size="7">
<p><span style="color: gray;">1. Population genomics with exome-seq</span></p>
</font><ol start="2" type="1"><font size="7">
<li><p>Massive phylogenetics with hyb-seq</p></li>
</font><li><font size="7"></font><p><font size="7">End-to-end R pipeline for metabolomics core </font></p></li>
</ol>
</section>
<section id="massive-phylogenetics-with-hyb-seq" class="slide level2">
<h2>2. <a href="https://www.biorxiv.org/content/10.1101/2022.07.31.502231v1">Massive phylogenetics with hyb-seq</a></h2>
<p><a href="https://github.com/HeatherKates/Nitfix-Recon/blob/main/Figures/Tree_Figure_NatureComm.png">Aim</a>: Produce the single largest <em>de novo</em> phylogenetic dataset constructed to date (<em>n</em> = 12,768 ingroup species)</p>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7910806/">Management of high throughput specimen data</a></p>
<ul>
<li><p>SQLite3 database to store and query project data</p></li>
<li><p>Transfer and Storage of hundreds of fastq.gz files weekly</p></li>
</ul>
<p><a href="https://www.frontiersin.org/articles/10.3389/fpls.2021.669064/full">NGS DNA sequence processing and assembly</a></p>
<p><a href="https://www.biorxiv.org/content/10.1101/2022.07.31.502231v1">Massively parallel phylogenetic analysis</a></p>
</section>
<section id="ngs-dna-sequence-processing-and-assembly" class="slide level2">
<h2>NGS DNA sequence processing and assembly</h2>
<p><font size="5"></font></p><font size="5">
<table>
<colgroup>
<col style="width: 38%">
<col style="width: 33%">
<col style="width: 27%">
</colgroup>
<thead>
<tr class="header">
<th>Process</th>
<th>Software</th>
<th>Independent executions</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Trimming and quality-filtering</td>
<td>Trimmomatic v. 0.39 (bash, hipergator)</td>
<td>14,769</td>
</tr>
<tr class="even">
<td>Quality control</td>
<td>FastQC (bash, hipergator)</td>
<td>14,769</td>
</tr>
<tr class="odd">
<td>Auxilary module: Calculate on-target reads:</td>
<td>Map to reference with BWA v. 0.7.17 + post-processing with samtools v. 1.10</td>
<td>14,769</td>
</tr>
<tr class="even">
<td>Contamination removal</td>
<td>aTRAM v. 2.0 (SPAdes wrapper) + custom BLAST script</td>
<td>14,769</td>
</tr>
<tr class="odd">
<td><em>De novo</em> contig assembly</td>
<td>SPAdes</td>
<td>14,769</td>
</tr>
<tr class="even">
<td>Custom gene assembly and paralog removal</td>
<td>Custom bash (aTRAM assembly, mafft alignment, FastTree phylogeny, and more)</td>
<td>100 genes x 13 data sets</td>
</tr>
</tbody>
</table>
</font><p><font size="5"></font></p>
</section>
<section id="custom-single-copy-gene-assembly-pipeline" class="slide level2">
<h2>Custom single-copy-gene assembly pipeline</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1"></a><span class="ex">[hkates@login9</span> SCN-gene-pipeline-git]$ ls</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="ex">Assemblies_to_SCN.bash</span>  LICENSE  README.md  Test_Data</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="ex">[hkates@login9</span> SCN-gene-pipeline-git]$ more Assemblies_to_SCN.bash </span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="co">#!/bin/bash</span></span>
<span id="cb1-5"><a href="#cb1-5"></a></span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="co">#The output of this program is $SUBANALYSIS_$LOCUS.single.copy.framed_contigsrr.fa and is a fasta file of all the single copy genes' contigs for a particular locus</span></span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="co">#And set of samples. This file can be aligned for gene tree building or concatenation or combined with same files from other sample sets.</span></span>
<span id="cb1-8"><a href="#cb1-8"></a></span>
<span id="cb1-9"><a href="#cb1-9"></a><span class="co">#Adjust the following variables for each locus and sampleset</span></span>
<span id="cb1-10"><a href="#cb1-10"></a><span class="va">LOCUS</span><span class="op">=</span>001</span>
<span id="cb1-11"><a href="#cb1-11"></a><span class="va">SUBANALYSIS</span><span class="op">=</span>Moraceae</span>
<span id="cb1-12"><a href="#cb1-12"></a><span class="va">DIR</span><span class="op">=</span>Test_Data/ <span class="co">#Name of directory where input files are</span></span>
<span id="cb1-13"><a href="#cb1-13"></a><span class="co">#Requirements to run:</span></span>
<span id="cb1-14"><a href="#cb1-14"></a><span class="co">#All de novo assembled contigs to be included in the analysis should be in a folder named Assemblies/ in DI</span></span>
<span id="cb1-15"><a href="#cb1-15"></a><span class="ex">R.</span> There can be assemblies from other subanalyses and loci </span>
<span id="cb1-16"><a href="#cb1-16"></a><span class="co">#In Assemblies/ as long as LOCUS is set and a file of the samples of interest is passed to -T</span></span>
<span id="cb1-17"><a href="#cb1-17"></a><span class="ex">module</span> load exonerate gcc/8.2.0 phyx mafft fasttree atram</span>
<span id="cb1-18"><a href="#cb1-18"></a><span class="co">#A file of sample names to be passed to -T in DIR.</span></span>
<span id="cb1-19"><a href="#cb1-19"></a><span class="co">#A reference sequence fasta in DIR. The name of the sequence (header) in the fasta file must match a string</span></span>
<span id="cb1-20"><a href="#cb1-20"></a> <span class="er">in</span> <span class="ex">the</span> name of the assemblies</span>
<span id="cb1-21"><a href="#cb1-21"></a></span>
<span id="cb1-22"><a href="#cb1-22"></a><span class="ex">atram_framer.py</span> <span class="at">-T</span> <span class="va">${DIR}${SUBANALYSIS}</span>_Taxa.txt <span class="at">-a</span> <span class="va">${DIR}</span>Assemblies/ <span class="at">-r</span> <span class="va">${DIR}</span>Locus_<span class="va">${LOCUS}</span>.fasta <span class="at">-o</span> <span class="va">${DI</span></span>
<span id="cb1-23"><a href="#cb1-23"></a><span class="er">R</span><span class="va">}</span>framed_contigs.<span class="va">${SUBANALYSIS}</span> <span class="at">-l</span> <span class="va">${DIR}</span>framer.Locus_<span class="va">${LOCUS}</span>.log <span class="at">-m</span> 350</span>
<span id="cb1-24"><a href="#cb1-24"></a><span class="co">#Impose a coverage cutoff to decide what sequences to drop from the framed contigs</span></span>
<span id="cb1-25"><a href="#cb1-25"></a><span class="co">#{1..n} where n = coverage cutoff, keeping in mind that spades header coverage is ~0.4X coverage (so n=2 is 4x coverage, roughly). change if desired.</span></span>
<span id="cb1-26"><a href="#cb1-26"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="dt">{</span><span class="dv">1</span><span class="dt">..</span><span class="dv">19</span><span class="dt">}</span><span class="kw">;</span> <span class="cf">do</span> <span class="fu">grep</span> c<span class="va">${i}</span>s <span class="va">${DIR}</span>framed_contigs.<span class="va">${SUBANALYSIS}</span>.Locus_<span class="va">${LOCUS}</span>.fasta <span class="op">&gt;&gt;</span> <span class="va">${DIR}${SUBANALYSIS}</span>.Drop_seqs.txt<span class="kw">;</span> <span class="cf">done</span></span>
<span id="cb1-27"><a href="#cb1-27"></a></span>
<span id="cb1-28"><a href="#cb1-28"></a><span class="co">#Use phyx to remove the low cov contigs</span></span>
<span id="cb1-29"><a href="#cb1-29"></a><span class="fu">sed</span> <span class="at">-i</span> <span class="st">'s/&gt;//g'</span> <span class="va">${DIR}${SUBANALYSIS}</span>.Drop_seqs.txt</span>
<span id="cb1-30"><a href="#cb1-30"></a><span class="ex">pxrms</span> <span class="at">-s</span> <span class="va">${DIR}</span>framed_contigs.<span class="va">${SUBANALYSIS}</span>.Locus_<span class="va">${LOCUS}</span>.fasta <span class="at">-f</span> <span class="va">${DIR}${SUBANALYSIS}</span>.Drop_seqs.txt <span class="at">-o</span> </span>
<span id="cb1-31"><a href="#cb1-31"></a><span class="va">${DIR}</span><span class="ex">framed_contigs.</span><span class="va">${SUBANALYSIS}</span><span class="ex">.Locus_</span><span class="va">${LOCUS}</span><span class="ex">.fasta.rms</span></span>
<span id="cb1-32"><a href="#cb1-32"></a></span>
<span id="cb1-33"><a href="#cb1-33"></a><span class="co">#Align framed contig with mafft and build tree with fasttree. Software can be changed.</span></span>
<span id="cb1-34"><a href="#cb1-34"></a><span class="ex">mafft</span> <span class="va">${DIR}</span>framed_contigs.<span class="va">${SUBANALYSIS}</span>.Locus_<span class="va">${LOCUS}</span>.fasta.rms <span class="op">&gt;</span> <span class="va">${DIR}</span>framed_contigs.Locus_<span class="va">${LOCUS}</span>.fa</span>
<span id="cb1-35"><a href="#cb1-35"></a><span class="ex">sta.rms.aln</span></span>
<span id="cb1-36"><a href="#cb1-36"></a><span class="fu">sed</span> <span class="at">-i</span> <span class="st">'/^&gt;/! s/n/-/g'</span> <span class="va">${DIR}</span>framed_contigs.Locus_<span class="va">${LOCUS}</span>.fasta.rms.aln</span>
<span id="cb1-37"><a href="#cb1-37"></a><span class="ex">pxclsq</span> <span class="at">-p</span> 0.1 <span class="at">-s</span> <span class="va">${DIR}</span>framed_contigs.Locus_<span class="va">${LOCUS}</span>.fasta.rms.aln <span class="at">-o</span> <span class="va">${DIR}</span>framed_contigs.Locus_<span class="va">${LOCUS}</span>.f</span>
<span id="cb1-38"><a href="#cb1-38"></a><span class="ex">asta.aln-cln</span></span>
<span id="cb1-39"><a href="#cb1-39"></a><span class="ex">FastTree</span> <span class="at">-nt</span> <span class="va">${DIR}</span>framed_contigs.Locus_<span class="va">${LOCUS}</span>.fasta.aln-cln <span class="op">&gt;</span> <span class="va">${DIR}</span>framed_contigs.Locus_<span class="va">${LOCUS}</span>.fasta.</span>
<span id="cb1-40"><a href="#cb1-40"></a><span class="ex">aln-cln.tre</span></span>
<span id="cb1-41"><a href="#cb1-41"></a></span>
<span id="cb1-42"><a href="#cb1-42"></a><span class="ex">module</span> load python/2.7.14 </span>
<span id="cb1-43"><a href="#cb1-43"></a><span class="co">#(older version of python is needed for the scripts below)</span></span>
<span id="cb1-44"><a href="#cb1-44"></a><span class="co">#Prune tree based on branch lengths (Yang and Smith)</span></span>
<span id="cb1-45"><a href="#cb1-45"></a><span class="co">#change relative-cutoff and absolute-cutoff after viewing tree</span></span>
<span id="cb1-46"><a href="#cb1-46"></a><span class="ex">python</span> /blue/soltis/share/Greg_HR_Nitfix/phylogenomic_dataset_construction/scripts/trim_tips.py <span class="va">${DIR}</span> tre </span>
<span id="cb1-47"><a href="#cb1-47"></a><span class="ex">5</span> .3</span>
<span id="cb1-48"><a href="#cb1-48"></a><span class="co">#Reduce clades that are all one taxon to a single tip (masking)</span></span>
<span id="cb1-49"><a href="#cb1-49"></a><span class="ex">python</span> /blue/soltis/share/Greg_HR_Nitfix/phylogenomic_dataset_construction/scripts/mask_tips_by_taxonID_tra</span>
<span id="cb1-50"><a href="#cb1-50"></a><span class="ex">nscripts.py</span> <span class="va">${DIR}</span> <span class="va">${DIR}</span> y</span>
<span id="cb1-51"><a href="#cb1-51"></a><span class="co">#Write fasta file for the tips that remain in the pruned tree</span></span>
<span id="cb1-52"><a href="#cb1-52"></a><span class="ex">python</span> /blue/soltis/share/Greg_HR_Nitfix/phylogenomic_dataset_construction/scripts/write_fasta_files_from_t</span>
<span id="cb1-53"><a href="#cb1-53"></a><span class="ex">rees.py</span> <span class="va">${DIR}</span>framed_contigs.Locus_<span class="va">${LOCUS}</span>.fasta.aln-cln <span class="va">${DIR}</span> mm <span class="va">${DIR}</span></span>
<span id="cb1-54"><a href="#cb1-54"></a></span>
<span id="cb1-55"><a href="#cb1-55"></a><span class="co">#Make csv of counts of how many contigs are left per sample</span></span>
<span id="cb1-56"><a href="#cb1-56"></a><span class="fu">grep</span> <span class="st">"&gt;"</span> <span class="va">${DIR}</span>framed_contigs.Locus_<span class="va">${LOCUS}</span>.fasta.aln-cln<span class="kw">|</span><span class="fu">cut</span> <span class="at">-d</span> <span class="st">"@"</span> <span class="at">-f</span> 1<span class="kw">|</span><span class="fu">sort</span><span class="kw">|</span><span class="fu">uniq</span> <span class="op">&gt;</span> <span class="va">${DIR}</span>Taxa.rr.txt</span>
<span id="cb1-57"><a href="#cb1-57"></a><span class="cf">while</span> <span class="bu">read</span> <span class="va">file</span><span class="kw">;</span> <span class="cf">do</span> <span class="bu">echo</span> <span class="at">-ne</span> Locus_<span class="va">${LOCUS}</span>,<span class="va">${file}</span>, <span class="kw">&amp;&amp;</span> <span class="fu">grep</span> <span class="at">-c</span> <span class="va">$file</span> <span class="va">${DIR}</span>framed_contigsrr.fa <span class="kw">;</span> <span class="cf">done</span> <span class="op">&lt;</span> <span class="va">${</span></span>
<span id="cb1-58"><a href="#cb1-58"></a><span class="va">DIR}</span>Taxa.rr.txt<span class="kw">|</span><span class="fu">sed</span> <span class="st">'s/&gt;//g'</span> <span class="op">&gt;</span> <span class="va">${DIR}${SUBANALYSIS}</span>_<span class="va">${LOCUS}</span>.Taxa_contigs_per_locus_counts.csv</span>
<span id="cb1-59"><a href="#cb1-59"></a></span>
<span id="cb1-60"><a href="#cb1-60"></a><span class="co">#Remove samples with more than one contigs from the alignment</span></span>
<span id="cb1-61"><a href="#cb1-61"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="kw">`</span><span class="fu">awk</span> <span class="at">-F</span> <span class="st">","</span> <span class="st">'{if ($3=="1") print $2;}'</span> <span class="va">${DIR}${SUBANALYSIS}</span>_<span class="va">${LOCUS}</span>.Taxa_contigs_per_locus_counts</span>
<span id="cb1-62"><a href="#cb1-62"></a><span class="ex">.csv</span><span class="kw">`</span></span>
<span id="cb1-63"><a href="#cb1-63"></a><span class="cf">do</span> <span class="fu">grep</span> <span class="at">-A1</span> <span class="va">$i</span> <span class="va">${DIR}</span>framed_contigsrr.fa<span class="kw">|</span><span class="fu">cut</span> <span class="at">-d</span> <span class="st">"@"</span> <span class="at">-f</span> 1 <span class="op">&gt;&gt;</span> <span class="va">${DIR}${SUBANALYSIS}</span>_<span class="va">${LOCUS}</span>.single.copy.frame</span>
<span id="cb1-64"><a href="#cb1-64"></a><span class="ex">d_contigsrr.fa</span><span class="kw">;</span> <span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="massively-parallel-phylogenetic-analysis" class="slide level2">
<h2>Massively parallel phylogenetic analysis</h2>
<p><font size="5"></font></p><font size="5">
<table>
<colgroup>
<col style="width: 38%">
<col style="width: 33%">
<col style="width: 27%">
</colgroup>
<thead>
<tr class="header">
<th>Process</th>
<th>Software</th>
<th>Independent executions</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Gene alignment</td>
<td>MAFFT v7.294b (bash, hipergator)</td>
<td>100 genes x 13 data sets</td>
</tr>
<tr class="even">
<td>Gene tree building</td>
<td>IQ-tree (bash, hipergator)</td>
<td>100 genes x 13 data sets</td>
</tr>
<tr class="odd">
<td>Supermatrix tree buiding</td>
<td>IQ-tree (bash, hipergator)</td>
<td>13 data sets</td>
</tr>
<tr class="even">
<td>Species tree estimation</td>
<td>ASTRAL-III (bash, hipergator)</td>
<td>13 data sets</td>
</tr>
<tr class="odd">
<td>Subtree scaffolding</td>
<td>custom bash script (hipergator)</td>
<td>NA</td>
</tr>
</tbody>
</table>
</font><p><font size="5"></font></p>
</section>
<section id="highlighted-bioinformatics-projects-3" class="slide level2">
<h2>Highlighted bioinformatics projects<br><br></h2>
<p><font size="7"></font></p><font size="7">
<p><span style="color: gray;">1. Population genomics with exome-seq</span></p>
<p><span style="color: gray;">2. Massive phylogenetics with hyb-seq</span></p>
</font><ol start="3" type="1"><font size="7">
</font><li><font size="7">End-to-end R pipeline for metabolomics core </font></li>
</ol>
</section>
<section id="end-to-end-r-pipeline-for-metabolomics-core" class="slide level2">
<h2>3. End-to-end R pipeline for metabolomics core</h2>
<p><a href="https://github.com/HeatherKates/SECIM_Reporting/blob/master/Example.Report.html">Aim</a>: Provide automated and flexible end-to-end analysis and reporting of metabolomics data</p>
<ul>
<li><p>Ingest datasets in multiple formats</p></li>
<li><p>Link all modules to enhance reproducibility</p></li>
</ul>
<p><font size="4"></font></p><font size="4">
<table>
<colgroup>
<col style="width: 38%">
<col style="width: 33%">
<col style="width: 27%">
</colgroup>
<thead>
<tr class="header">
<th>Process</th>
<th>Software</th>
<th>Independent executions</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Missing data imputation and normalization</td>
<td>Metaboanalyst (R, hipergator)</td>
<td>As needed</td>
</tr>
<tr class="even">
<td>Multivariate Statistics</td>
<td>car (R)</td>
<td>As needed</td>
</tr>
<tr class="odd">
<td>Univariate Statistics</td>
<td>stats (R), lmer (R), emmeans (R)</td>
<td>As needed</td>
</tr>
<tr class="even">
<td>KEGG subnetwork estimation</td>
<td>FELLA (R)</td>
<td>As needed</td>
</tr>
<tr class="odd">
<td>Visualizations</td>
<td>plotly (R)</td>
<td>As needed</td>
</tr>
<tr class="even">
<td>Reporting method</td>
<td>R markdown (R); html output</td>
<td>As needed</td>
</tr>
</tbody>
</table>
</font><p><font size="4"></font></p>
<div class="footer footer-default">

</div>
</section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="HOBI_Bioinfo_Presentation_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="HOBI_Bioinfo_Presentation_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="HOBI_Bioinfo_Presentation_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="HOBI_Bioinfo_Presentation_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="HOBI_Bioinfo_Presentation_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="HOBI_Bioinfo_Presentation_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="HOBI_Bioinfo_Presentation_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="HOBI_Bioinfo_Presentation_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="HOBI_Bioinfo_Presentation_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="HOBI_Bioinfo_Presentation_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'smaller': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: false,

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        text: function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn) {
        const config = {
          allowHTML: true,
          content: contentFn,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        };
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>